### DISCUSSION TITLE
After setting fp_task_min, e.g. =20, the fp data for one iter is less than 20 and training step is skiped. However, after several iters, these fp data are many and want to include them. Is there a setting to do that?

### DISCUSSION THREAD

**Discussion by roger13231** [deepmodeling/dpgen#1653] [Q&A]
As described in title. I have several iterations and each of them have around 10-15 fp data. The accurate is > 99% and fp data is few, so the training step is skipped which is expected. However, after several iterations, there are many of them and want to include them.

I found that in later training step (which is not skipped), these fp data is not included. I tried to maually change the input.json in one training step, but it goes back to normal (without these fp data) in the later iteration training step. 

I am wondering whether there is a way to include them in later iterations.

**Comment by njzjz** [comment]
> these fp data is not included

This is not expected. Are you sure `iter.*/data.*` exists?

In the code, all previous iterations should be included.

https://github.com/deepmodeling/dpgen/blob/47a23b393c6e071bc588a2171780f09863741d3a/dpgen/generator/run.py#L394-L398

**Comment by roger13231** [comment]
Hi, Jinzhe,

Thanks for you rapid reply. The iter.*/data.* exists.

I attached a few figures. fp_task_min is set to 20, and max is set to 80. 
1. iter68 fp have system5 more than 20. The others have data less than 20;
2. Iter68 02.fp folder have data.003-007, 015, and 016. I checked these folder, all data folder have 'set.' folder with unempty data;
3. Iter62. All system fp data is less than 20..
4. Iter62. All 02.fp folder has data. folder having 'set.' folder wtih unempty data;
5. In the input.json file for training step in iter69, iter68 have only system5 data, and no iter62.

So, it seems like even data folder is generated, if the data is less than fp_task_min, the training does not include these data.
![data1](https://github.com/user-attachments/assets/3fdc56be-dee3-46dc-955a-393f9792109a)
![data1_1](https://github.com/user-attachments/assets/4930a691-a223-4ccb-993e-7619058d6007)
![data2](https://github.com/user-attachments/assets/a76dc9e4-61e4-4cfa-8d1c-3af1eb4092c0)
![data2_1](https://github.com/user-attachments/assets/40444e72-7ad7-4ac1-9a98-5059aca0c86a)
![input](https://github.com/user-attachments/assets/2afe17f5-ddd7-4f3e-a4f0-72335ec77f10)

**Comment by njzjz** [comment]
Okay I found the related codes:

https://github.com/deepmodeling/dpgen/blob/47a23b393c6e071bc588a2171780f09863741d3a/dpgen/generator/run.py#L418-L422

which was added in https://github.com/deepmodeling/dpgen/commit/37e59f8f5f7540d67af0d832d4737b81979bf800

@wanghan-iapcm could you clarify this behavior? It's confusing that these data was calculated but not used. By the way, this behavior is out of my expectations.
